src/wp-includes/class-wp-block-type.php

```diff
@@ -368,6 +378,18 @@ public function set_props( $args ) {
 		foreach ( $args as $property_name => $property_value ) {
 			$this->$property_name = $property_value;
 		}
+
+		// Setup attributes if needed.
```

Comment:
Maybe this logic should be executed before `register_block_type_args` so plugin authors could access it when running their filters?

Reply:
Interesting point. That would allow plugin authors to remove the attribute. Do we want that with the built-in attributes?

Reply:
> That would allow plugin authors to remove the attribute. Do we want that with the built-in attributes?

I can see a use case for this, mainly in conjunction with another filter to enforce locking rather than disable it.